Library/TeamTalkLib/bin/ttsrv/ServerXML.cpp

```diff
@@ -1493,6 +1493,44 @@ namespace teamtalk{
         AddNewUser(updateduser);
     }
 
+bool ServerXML::CleanupChannelOperators(int deletedChannelID)
+{
+    bool overallModified = false;
+    int userIndex = 0;
+    UserAccount currentUser;
+
+    std::vector<UserAccount> usersToModify;
+    std::vector<std::string> usernamesToRemove;
+
+    userIndex = 0;
+    currentUser = UserAccount();
+    while (GetNextUser(userIndex++, currentUser))
+    {
+        if (currentUser.auto_op_channels.count(deletedChannelID))
+        {
+            usernamesToRemove.push_back(UnicodeToUtf8(currentUser.username).c_str());
+            currentUser.auto_op_channels.erase(deletedChannelID);
+            usersToModify.push_back(currentUser);
+            overallModified = true;
+        }
+        currentUser = UserAccount();
+    }
+
+    if (overallModified)
+    {
+        for (const auto& username : usernamesToRemove)
```

Comment:
You don't need `usernamesToRemove` vector. Just use run the loop below like this:
```
        for (const auto& userAcc : usersToModify)
        {
            RemoveUser(UnicodeToUtf8(userAcc.username).c_str()););
            AddNewUser(userAcc);
        }
```